techmatters/flex-plugins
# CHI-1176: Fix prepopulated date

Primary reviewer: @mmythily 

## Description

* Fixes issue where the initial date on date-input form items is incorrect in time zones with a '+' offset
* Added test coverage for getInitialValue in formGenerator.tsx
* Demonstrates how to test date logic deterministically in Jest.

### Checklist
- [ X ] Corresponding issue has been opened
- [ X ] New tests added
- [ N/A ] Strings are localized
- [ X  ] Tested for chat contacts
- [ NO ] Tested for call contacts

### Related Issues
Fixes CHI-1176

### Verification steps

* Set your system timezone to a positive offset.
* Open a form with a date that prepopulates to today (like the one in Offline Contacts)
* Check the initial value is today's date (as in, today for the timezone you just set :-) )

Repeat the above steps with the system timezone set to a negative offset
